functions/src/business-logics/share-logics.ts

```diff
@@ -1,4 +1,80 @@
+import {Action, LogicConfig, LogicResultDoc} from "emberflow/lib/types";
+import {Entity} from "../db-structure";
+import {db} from "emberflow/lib";
 // TODO write onShareLogic
 //  copy post to user's timeline
 //  increment post sharesCount
 //  create a notification for post's author
+
+
+export const onShareLogic: LogicConfig = {
```

Comment:
Additional task: create another logic result doc that will add the userview of the user in the "posts/postId/share/[userId]". you can expect the docpath to have this path so you can just use it.